R/process_data.R

```diff
@@ -137,79 +172,48 @@ center_ipd <- function(ipd, agd){
 
 
 
+# Functions NOT to be exported ---------------------------------------
 
 
-
-
-
-
-
-#' helper function: transform TTE ADaM data to suitable input for survival R pkg
-#'
-#' @param dd data frame, ADTTE read via `haven::read_sas()`
-#' @param time_scale a character string, 'year', 'month', 'week' or 'day', time unit of median survival time
-#'
-#' @return a data frame can be used as input to [survival::Surv()]
-ext_tte_transfer <- function(dd, time_scale = "month", trt = NULL) {
-  timeUnit <- list("year" = 365.24, "month" = 30.4367, "week" = 7, "day" = 1)
-
-  if (!time_scale %in% names(timeUnit)) stop("time_scale has to be 'year', 'month', 'week' or 'day'")
-
-  if ("CENSOR" %in% names(dd)) {
-    dd <- dd[!is.na(dd$CENSOR), ]
-    dd$status <- as.numeric(dd$CENSOR == 0)
-  }
-  if ("EVENT" %in% names(dd)) {
-    dd$status <- as.numeric(as.character(dd$EVENT))
-  }
-  if ("TIME" %in% names(dd)) {
-    dd$AVAL <- as.numeric(as.character(dd$TIME))
-  }
-
-  dd$time <- dd$AVAL * timeUnit[[time_scale]]
-  if (!is.null(trt)) dd$treatment <- trt
-  as.data.frame(dd)
-}
-
-
-
-# functions NOT to be exported ---------------------------------------
-
 #' Calculate pooled arm statistics in Aggregated Data (AgD) based on arm-specific statistics
 #'
-#' @param use_agd
+#' This is a convenient function to pool arm statistics ## FILL IN
```

Comment:
@MikeJSeo @hoppanda Could we improve the docs here with a return description and an example?
Then we can resolve the question about where `use_agd` comes from

Reply:
Here is a suggestion. Feel free to change!

#' Calculate pooled arm statistics in Aggregated Data (AgD) based on arm-specific statistics
#'
#' This is a convenient function to pool arm statistics. This function is called
#' within process_agd and when the ARM is not equal to "Total". Note pooled
#' median can't be calculated and it is only an approximation.
#'
#' @param use_agd aggregated data that is processed within process_agd
#'
#' @return Complete N, count, mean, sd, and median for the pooled arm